docs/Sources/tuist/tuist.docc/Articles/Tuist/Extensions/generation-environment.md

```diff
@@ -0,0 +1,49 @@
+# Generation-time configuration
+
+Learn how to leverage environment variables that can be read from the manifest files.
+
+There are certain scenarios where you might need to customize the projects at generation time. For example, if you are building a white-label app, you might want to use the same project structure, but adjust some attributes in a per-app basis. For example, the name of the app.
+
+### Using Tuist environment variables
+
+To facilitate that, Tuist allows passing configuration through environment variables that can be accessed from the manifest files. For example:
+
+```bash
+TUIST_APP_NAME=MyApp tuist generate
+```
+
+If you want to pass multiple environment variables just separate them with a space. For example:
+
+```bash
+TUIST_APP_NAME=MyApp TUIST_APP_LOCALE=pl tuist generate
+```
+
+Variables can be accessed using the `Environment` type. Any variables following the convention `TUIST_XXX` defined in the environment or passed to Tuist when running commands will be accessible using the `Environment` type.
+The following example shows how we access the `TUIST_APP_NAME` variable:
+
+```swift
+func appName() -> String {
+    if case let .string(environmentAppName) = Environment.appName {
+        return environmentAppName
+    } else {
+        return "MyApp"
+    }
+}
+```
+
+Accessing variables returns an instance of type `Environment.Value?` which can take any of the following values:
+
+| Case       | Description                                                                                                                                                     |
+| ---------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `.boolean` | Used when the variable represents a boolean. The values 1, true, TRUE, yes and YES are casted to `true`, and 0, false, FALSE, no, and NO are casted to `false`. |
```

Comment:
We can remove this one. It doesn't exist [anymore](https://github.com/tuist/tuist/blob/1e14600ccb545aeded122b35d512f2fa8f84db80/Sources/ProjectDescription/Environment.swift#L7)